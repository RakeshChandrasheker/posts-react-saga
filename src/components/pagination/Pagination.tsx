import { FunctionComponent } from "react"
import './Pagination.scss';

export interface IPagination {
  postsPerPage: number;
  totalPosts: number;
  paginate: (number: number) => void;
}

const Pagination: FunctionComponent<IPagination> = ({ postsPerPage, totalPosts, paginate }): JSX.Element => {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalPosts / postsPerPage); i++) {
    pageNumbers.push(i);
  }

  return (
    <ul className='pagination'>
      {pageNumbers.map(number => (
        <li key={number} className='page-item'>
          <a onClick={() => paginate(number)} href='!#' className='page-link'>
            {number}
          </a>
        </li>
      ))}
    </ul>
  )
}

export default Pagination
