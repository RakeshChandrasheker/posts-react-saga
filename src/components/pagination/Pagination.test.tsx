import { shallow } from 'enzyme';
import Pagination, { IPagination } from './Pagination';

describe('<Pagination />', () => {

  const paginationProps: IPagination = {
    postsPerPage: 122,
    totalPosts: 112,
    paginate: jest.fn()
  }

  const wrapper = shallow(<Pagination {...paginationProps} />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should simulate paginate()', () => {
    wrapper.find('a').simulate('click');
    expect(paginationProps.paginate).toHaveBeenCalledTimes(1)
  });

  it('should Match Snapshot', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  afterAll(() => {
    wrapper.unmount();
  });
});
