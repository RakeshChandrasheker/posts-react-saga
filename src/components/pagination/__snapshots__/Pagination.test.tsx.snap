// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should Match Snapshot 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="!#"
      onClick={[Function]}
    >
      1
    </a>
  </li>
</ul>
`;
